Userland/Services/SystemServer/main.cpp

```diff
@@ -482,6 +482,28 @@ static ErrorOr<void> create_tmp_semaphore_directory()
     return {};
 }
 
+static ErrorOr<void> activate_services(Core::ConfigFile const& config)
+{
+    for (auto const& name : config.groups()) {
+        auto service = TRY(Service::try_create(config, name));
+        if (service->is_enabled_for_system_mode(g_system_mode)) {
+            TRY(service->setup_sockets());
+            g_services.append(move(service));
+        }
+    }
+    // After we've set them all up, activate them!
+    dbgln("Activating {} services...", g_services.size());
+    for (auto& service : g_services) {
+        dbgln_if(SYSTEMSERVER_DEBUG, "Activating {}", service->name());
+    }
+
+    for (auto& service : g_services) {
```

Comment:
Why are you looping over `g_services` twice? That `dbgln_if` above should probably be added here instead.